.github/workflows/gosec.yml

```diff
@@ -0,0 +1,26 @@
+name: Run Gosec
+on:
+  workflow_dispatch:
+  schedule:
+    #        ┌───────────── minute (0 - 59)
+    #        │  ┌───────────── hour (0 - 23)
+    #        │  │ ┌───────────── day of the month (1 - 31)
+    #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+    #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        │  │ │ │ │
+    #        *  * * * *
+    - cron: '30 2 * * *'
```

Comment:
Why not running this for every PR?

Reply:
I didn’t want it to be a blocker for development for now. This workflow fails when it finds potential issues, some of which may also be false positives. Right now there are a lot of issues that it flags so it would always be failing.

Reply:
I think we need to start fixing the issues, otherwise this will be useless. We do have it enabled in the core collector and was not that hard :)

Reply:
Where is it enabled in the core collector? I opened this same PR in the core collector repo because I thought it did not exist already. Even in the core collector this GoSec workflow fails because of many issues.

I will also create a new issue to fix GoSec issues if this PR gets merged. I think it would be best to fix the initial set of issues in a separate PR and then enable this for every future PR.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/blob/d885cb20e11c5b8c9dafa3251b2591cc8f1f8c82/.golangci.yml

Reply:
@bogdandrutu looks like we restricted gosec to a very small set of rules in the collector, I will follow up and update this to run the same rules at the core collector and run on PR builds as well

Reply:
> small set of rules in the core collector

We disable only 2 rules "G402" and "G404" :)

Reply:
Ah nevermind, I thought those were the only 2 being used

Reply:
Please start fixing problems if you want, see https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1955 that enables always gosec